### ISSUE TITLE
Displayed height of nested amp-img is erroneous

### ISSUE THREAD

**Issue by medicalspanw** [ampproject/amphtml#10085]
The algorithm for displaying the height of an image from a nested amp-img structure is not working correctly. 

## How do we reproduce the issue?
```html
<amp-img src="some square webp file" height='Ho' width='Wo' layout='responsive'>
  <amp-img src="some square jpg file" height='Hi' width='Wi' layout='responsive'></amp-img>
</amp-img>
```

(My initial comment included white space to make the scenarios look like a table. Below is revised without the white space.)

The following numbers represent the following: a scenario, the setting for Ho, the setting for Wo, the setting for Hi, the setting for Wi, the height and width of the displayed image.

S Ho Wo Hi Wi HxW
1 50 50 50 50 100x50
2 25 50 25 50 50x50
3 0 50 50 50 50x50
4 50 0 50 50 50x50
5 50 50 0 50 50x50
6 50 50 50 0 50x50
7 0 0 50 50 50x50
8 50 50 0 0 50x50
9 0 0 0 0 0x0
scenario 9 is correct. 

## What browsers are affected?
Version 59.0.3071.104 (openSUSE Build) (64-bit). No other browsers have been tested.

**Comment by jridgewell** [comment]
These are responsive images, so how wide is the container?

**Comment by medicalspanw** [comment]
Thank you for your promt reply.

I uploaded the file to www.physmed.biz/index.html.

Note that the links probably will not work.

Thanks again.

**Comment by jridgewell** [comment]
I don't understand what you're trying to do here.

**Comment by medicalspanw** [comment]
Using the Google web site grading tool, the web gets dinged for not using
webp files. However, Safari does not support this format. The solution is
to have webp files delivered for browsers that support it, and jpg files
for browsers that don't. Hence, I've nested amp-img elements.

It turns out that there are some other subtle bugs with the way Safari
renders. The logo image rendered correctly. Images located in the
amp-carousel do not.

My work around is to abandon the nested amp-img elements and take the hit
on not supplying webp files.

Thanks.


BTW: The code that I posted has been modified with the new approach.
Thanks, again.
